src/app/http.rs

```diff
@@ -0,0 +1,73 @@
+use axum::{
+    body::Body,
+    routing::{get, post},
+    Extension, Router,
+};
+use http::{
+    header::{self, HeaderName},
+    Method, Response,
+};
+use std::{net::SocketAddr, sync::Arc};
+use tower_http::cors::{Any, CorsLayer};
+use tracing::error;
+
+use super::{config::AppConfig, context::AppContext, endpoints};
+
+pub fn build_router(context: Arc<AppContext>, maxmind: Arc<maxminddb::Reader<Vec<u8>>>) -> Router {
+    let cors = CorsLayer::new()
+        .allow_methods([Method::GET, Method::POST])
+        .allow_headers([
+            header::AUTHORIZATION,
+            header::CACHE_CONTROL,
+            header::CONTENT_LENGTH,
+            header::CONTENT_TYPE,
+            header::IF_MATCH,
+            header::IF_MODIFIED_SINCE,
+            header::IF_NONE_MATCH,
+            header::IF_UNMODIFIED_SINCE,
+            header::RANGE,
+            header::HeaderName::from_static("x-request-type"),
+            HeaderName::from_static("x-agent-label"),
```

Comment:
```suggestion
            HeaderName::from_static("X-Request-Type"),
            HeaderName::from_static("X-Agent-Label"),
```
Обычно заголовки пишут с большой буквы или здесь специально так написано?

Reply:
https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2

> Just as in HTTP/1.x, header field names are strings of ASCII
> characters that are compared in a case-insensitive fashion.  However,
> header field names MUST be converted to lowercase prior to their
> encoding in HTTP/2.  A request or response containing uppercase
> header field names MUST be treated as malformed.

Интернет изменился в последние годы :)